e2e-tests/playwright/tests/functional/channels/edit_file_attachments/edit_file_attachment.spec.ts

```diff
@@ -0,0 +1,372 @@
+    await channelPage.centerView.postEdit.toBeVisible();
+    await channelPage.centerView.postEdit.removeFile('sample_text_file.txt');
+    await channelPage.centerView.postEdit.writeMessage('');
+    await channelPage.centerView.postEdit.sendMessage();
+
+    await channelPage.centerView.postEdit.deleteConfirmationDialog.toBeVisible();
+    await channelPage.centerView.postEdit.deleteConfirmationDialog.confirmDeletion();
+    await channelPage.centerView.postEdit.deleteConfirmationDialog.notToBeVisible();
+
+    expect(channelPage).not.toContain(originalMessage);
+    expect(channelPage).not.toContain('sample_text_file.txt');
+});
+
+test('MM-T5656_1 should be able to restore previously edited post version that contains attachments', async ({pw}) => {
+    test.setTimeout(120000);
+
+    const originalMessage = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit';
+    const newMessage = 'New Message';
+
+    const {user} = await pw.initSetup();
+    const {page} = await pw.testBrowser.login(user);
+    const channelPage = new pages.ChannelsPage(page);
+
+    await channelPage.goto();
+    await channelPage.toBeVisible();
+    await channelPage.centerView.postCreate.postMessage(originalMessage, ['sample_text_file.txt']);
+
+    const post = await channelPage.centerView.getLastPost();
+    await post.toBeVisible();
+    await post.toContainText(originalMessage);
+    await post.toContainText('sample_text_file.txt');
+
+    await post.hover();
+    await post.postMenu.toBeVisible();
+    await post.postMenu.dotMenuButton.click();
+
+    await channelPage.postDotMenu.toBeVisible();
+    await channelPage.postDotMenu.editMenuItem.click();
+    await channelPage.centerView.postEdit.toBeVisible();
+    await channelPage.centerView.postEdit.removeFile('sample_text_file.txt');
+    await channelPage.centerView.postEdit.writeMessage(newMessage);
+    await channelPage.centerView.postEdit.sendMessage();
+
+    const updatedPost = await channelPage.centerView.getLastPost();
+    await updatedPost.toBeVisible();
+    await updatedPost.toContainText(newMessage);
+    expect(updatedPost).not.toContain('sample_text_file.txt');
+
+    const postID = await channelPage.centerView.getLastPostID();
+    await channelPage.centerView.clickOnLastEditedPost(postID);
+
+    await channelPage.sidebarRight.toBeVisible();
+    await channelPage.sidebarRight.verifyCurrentVersionPostMessage(postID, newMessage);
+
+    await channelPage.sidebarRight.restorePreviousPostVersion();
+
+    await channelPage.centerView.postEdit.restorePostConfirmationDialog.toBeVisible();
+    await channelPage.centerView.postEdit.restorePostConfirmationDialog.confirmRestore();
+    await channelPage.centerView.postEdit.restorePostConfirmationDialog.notToBeVisible();
+
+    const restoredPost = await channelPage.centerView.getLastPost();
+    await restoredPost.toBeVisible();
+    expect(restoredPost.toContainText('sample_text_file.txt'));
+});
```

Comment:
In general, please add comments about the steps and verifications to easily map/verify if it matches the one in written test case. Would be helpful especially for non-tech reviewer like QA. TIA.

Reply:
Hi @saturninoabril 
https://github.com/mattermost/mattermost-test-management/pull/201
I've upadted the tests scenarios in zephry  https://github.com/mattermost/mattermost-test-management/pull/201 and testkeys in this PR.